api/core/key.go

```diff
@@ -147,11 +166,50 @@ func (k Key) Uint(v uint) KeyValue {
 	return k.Uint64(uint64(v))
 }
 
+func (k Key) Struct(v interface{}) KeyValue {
+	return KeyValue{
+		Key: k,
+		Value: Value{
+			Type:   STRUCT,
+			Struct: v,
+		},
+	}
+}
+
+func (k Key) Encode(v func(io.Writer) error) KeyValue {
+	return KeyValue{
+		Key: k,
+		Value: Value{
+			Type:    ENCODER,
+			Encoder: v,
+		},
+	}
+}
+
 func (k Key) Defined() bool {
 	return k.Variable.Defined()
 }
 
-// TODO make this a lazy one-time conversion.
+func (v Value) Evaluate() Value {
```

Comment:
The issue of making a copy of the `Value` object was brought up in the #59 too. So just to keep track of that I'll mention it here: the `BYTES` case should also be handled here by making a new slice and copying a contents of the original slice into the new slice.